neo/app/assets/javascripts/index.js

```diff
@@ -90,15 +90,246 @@ app.controller('SearchCtrl', function($scope, $window, Asterix) {
+      labels: []
+    },
+    status: {
+      init: true,
+      zoomLevel: 4,
+      logicLevel: 'state'
+    },
+    styles: {
+      initStyle: {
+        weight: 2,
+        fillOpacity: 0.5,
+        color: 'white'
+      },
+      stateStyle: {
+        fillColor: '#f7f7f7',
+        weight: 2,
+        opacity: 1,
+        color: '#92c5de',
+        dashArray: '3',
+        fillOpacity: 0.2
+      },
+      countyStyle: {
+        fillColor: '#f7f7f7',
+        weight: 1,
+        opacity: 1,
+        color: '#92c5de',
+        fillOpacity: 0.2
+      },
+      hoverStyle: {
+        weight: 5,
+        color: '#666',
+        dashArray: '',
+        fillOpacity: 0.7
+      }
+    },
+
+  });
+
+  // initialize
+  $scope.init = function () {
+    setInfoControl();
+    loadGeoJsonFiles();
+  };
+
+  function setInfoControl() {
+    // Interaction function
+    function highlightFeature(feature, leafletEvent) {
+      var layer = leafletEvent.target;
+      layer.setStyle($scope.styles.hoverStyle);
+      layer.bringToFront();
+      $scope.selectedPlace = feature;
+    }
+    
+    function resetHeight(leafletEvent) {
+      var style;
+      if (!$scope.status.init)
+        style = {
+          weight: 2,
+          fillOpacity: 0.5,
+          color: 'white'
+        };
+      else
+        style = {
+          weight: 1,
+          fillOpacity: 0.2,
+          color: '#92c5de'
+        }
+      if(leafletEvent)
+        leafletEvent.target.setStyle(style);
+    }
+
+    function zoomToFeature(leafletEvent) {
+      leafletData.getMap().then(function(map) {
+        if(leafletEvent)
+          map.fitBounds(leafletEvent.target.getBounds());
+      });
+    }
+
+    $scope.$on("leafletDirectiveGeoJson.mouseover",function(ev, leafletPayload){
+      highlightFeature(leafletPayload.leafletObject.feature, leafletPayload.leafletEvent);
+    });
+
+    $scope.$on("leafletDirectiveGeoJson.mouseout",function(ev, leafletPayload){
+      resetHeight(leafletPayload.leafletEvent);
+    });
+
+    $scope.$on("leafletDirectiveGeoJson.click",function (ev, leafletPayload) {
+      zoomToFeature(leafletPayload.leafletEvent);
+    });
+
+    // add info control
+    var info = L.control();
+
+    info.onAdd = function () {
+      this._div = L.DomUtil.create('div', 'info'); // create a div with a class "info"
+      this._div.innerHTML = [
+        '<h4>Count by {{ status.logicLevel }}</h4>',
+        '<b>{{ selectedPlace.properties.name || "No place selected" }}</b>',
+        '<br/>',
+        'Count: {{ selectedPlace.properties.count || "0" }}'
+      ].join('');
+      $compile(this._div)($scope);
+      return this._div;
+    };
+
+    info.options = { position: 'topleft' };
+    $scope.controls.custom.push(info);
+  }
+
+  // load geoJson
+  function loadGeoJsonFiles() {
+    $http.get("assets/data/state.json")
+      .success(function (data) {
+        $scope.geojson.state = {
+          data: data,
```

Comment:
be careful that this request is asynchronized, so the data may not ready when the query issues.
do you have any corresponding logics?